src/main/java/hu/wtomee/objects/DFA.java

```diff
@@ -84,6 +86,17 @@ public void checkTransitions() {
                 throw new InvalidParameterException("Symbol doesnt exists in transition: " + t.toString());
             }
 
+            // Check if deterministic
+            int counter = 0;
+            for (Transition t2 : transitions) {
+                if (t.symbol.equals(t2.symbol) && t.fromState.equals(t2.fromState)) {
+                    if (t.nextState.equals(t2.nextState) && counter < 2) {
+                        counter++;
+                    } else {
+                        throw new InvalidParameterException("DFA is a not deterministic because of these transitions:" + t + "\n" + t2);
+                    }
```

Comment:
Ez nincs teljesen renddben. Most akkor throwolsz, ha talalsz egy teljesen ugyanolyan transitiont. Tehat ha van mondjuk ket `(p,a,q)`. Tehat nem a nemdeterminisztikussagra throwolsz igazabol, hanem arra, ha van duplikatum a transitions-ben. 

Amire viszont nem throw-olsz, pedig kellene, az az, ha van `(p,a,q)` es `(p,a,q2)` transition is.

Reply:
khm... unit testek ;-)